### PR TITLE
Updated cohesity checks to be more compliant to Checkmk 2.0

### PR DESCRIPTION
**Pull request by sritd**
Updated cohesity checks and package to be more compatibel to Checkmk 2.0.

Signed-off-by: Sven Rueß <github@sritd.de>

### CODE REVIEW

File: cohesity/src/agents/special/agent_cohesity [hunk 1]
```diff
@@ -48,8 +48,8 @@ def get_header():
     try:
         response = requests.post(
             url, data=creds, headers=header, verify=False)
-    except APIException as exp:
```
Comment (Bastian-Kuhn) [inline cohesity/src/agents/special/agent_cohesity:51]: @sritd I think we should catch both exceptions here: except (APIException, OSError) as exp. Wondering in which case we have an OSError?

File: cohesity/src/agents/special/agent_cohesity [hunk 2]
```diff
@@ -70,8 +70,8 @@ def get_url(url):
             API_URL + url,
             headers=HEADER,
             verify=False)
-    except APIException as exp:
-        print(f"Cannot get Clusterstatus {exp}")
```
Comment (Bastian-Kuhn) [inline cohesity/src/agents/special/agent_cohesity:74]: Same here

File: cohesity/src/info [hunk 3]
```diff
@@ -1,18 +1,18 @@
 {'author': 'Sven Rueß, sritd.de, Bastian Kuhn, bastian-kuhn.de',
- 'description': 'Cohesity checks',
+ 'description': 'Cohesity checks\n',
```
Comment (Bastian-Kuhn) [inline cohesity/src/info:2]: Don't understand the reason for the line break here? Is it cause of CMK 2.0?

**Review by Bastian-Kuhn — CHANGES_REQUESTED** [review]
Just some questions and comments

File: cohesity/src/agents/special/agent_cohesity (hunk 1 above)
Comment (sritd) [reply cohesity/src/agents/special/agent_cohesity:51]: APIException is not defined. Missing import?

File: cohesity/src/agents/special/agent_cohesity (hunk 2 above)
Comment (sritd) [reply cohesity/src/agents/special/agent_cohesity:74]: Same like above. Where is this exception defined?

File: cohesity/src/info (hunk 3 above)
Comment (sritd) [reply cohesity/src/info:2]: Yes, This is done by Checkmk directly. I configured it in GUI.

File: cohesity/src/agents/special/agent_cohesity (hunk 1 above)
Comment (sritd) [reply cohesity/src/agents/special/agent_cohesity:51]: OSError is thrown, if connection can not be established.

Comment (Bastian-Kuhn) [reply cohesity/src/agents/special/agent_cohesity:51]: OK, APIException is left from the SDK (which is not used anymore and therefore wrong, thank you for removing). But OSError is not a requests exception, just the base for the base Exceptions in the Requests module: https://docs.python-requests.org/en/master/_modules/requests/exceptions/.

So general would be to catch RequestException or ConnectionError?

Comment (sritd) [reply cohesity/src/agents/special/agent_cohesity:51]: That is not enough. OSError is thrown, if DNS can not be resolved or connection is not possible. That is not done by requests module. That is done by socket. I think, this is correct.

**Review by Bastian-Kuhn — APPROVED** [review]

**Pull request merged**